### DISCUSSION TITLE
ModuleNotFoundError: No module named 'groundingdino'

### DISCUSSION THREAD

**Discussion by BNP1111** [lllyasviel/Fooocus#3289] [Q&A] [answered]
D:\Fooocus-2>.\python_embeded\python.exe -s Fooocus\entry_with_update.py --preset realistic
Update failed.
unsupported URL protocol
Update succeeded.
[System ARGV] ['Fooocus\\entry_with_update.py', '--preset', 'realistic']
Loaded preset: D:\Fooocus-2\Fooocus\presets\realistic.json
Failed to load config key: {"path_checkpoints": "D:\\Fooocus\\Fooocus\\models\\checkpoints"} is invalid or does not exist; will use {"path_checkpoints": ["../models/checkpoints/"]} instead.
Failed to load config key: {"path_loras": "D:\\Fooocus\\Fooocus\\models\\loras"} is invalid or does not exist; will use {"path_loras": ["../models/loras/"]} instead.
Failed to load config key: {"path_embeddings": "D:\\Fooocus\\Fooocus\\models\\embeddings"} is invalid or does not exist; will use {"path_embeddings": "../models/embeddings/"} instead.
Failed to load config key: {"path_vae_approx": "D:\\Fooocus\\Fooocus\\models\\vae_approx"} is invalid or does not exist; will use {"path_vae_approx": "../models/vae_approx/"} instead.
Failed to load config key: {"path_upscale_models": "D:\\Fooocus\\Fooocus\\models\\upscale_models"} is invalid or does not exist; will use {"path_upscale_models": "../models/upscale_models/"} instead.
Failed to load config key: {"path_inpaint": "D:\\Fooocus\\Fooocus\\models\\inpaint"} is invalid or does not exist; will use {"path_inpaint": "../models/inpaint/"} instead.
Failed to load config key: {"path_controlnet": "D:\\Fooocus\\Fooocus\\models\\controlnet"} is invalid or does not exist; will use {"path_controlnet": "../models/controlnet/"} instead.
Failed to load config key: {"path_clip_vision": "D:\\Fooocus\\Fooocus\\models\\clip_vision"} is invalid or does not exist; will use {"path_clip_vision": "../models/clip_vision/"} instead.
Failed to load config key: {"path_fooocus_expansion": "D:\\Fooocus\\Fooocus\\models\\prompt_expansion\\fooocus_expansion"} is invalid or does not exist; will use {"path_fooocus_expansion": "../models/prompt_expansion/fooocus_expansion"} instead.
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
Fooocus version: 2.5.0-rc1
Version mismatch for tqdm: Installed version 4.64.1 does not meet requirement tqdm==4.65.0
Installing requirements
Couldn't install requirements.
Command: "D:\Fooocus-2\python_embeded\python.exe" -m pip install -r "requirements_versions.txt" --prefer-binary
Error code: 1
stdout: Looking in indexes: https://pypi.org/simple, https://pypi.ngc.nvidia.com
Requirement already satisfied: torchsde==0.2.5 in d:\fooocus-2\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 1)) (0.2.5)
Requirement already satisfied: einops==0.4.1 in d:\fooocus-2\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 2)) (0.4.1)
Requirement already satisfied: transformers==4.30.2 in d:\fooocus-2\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 3)) (4.30.2)
Requirement already satisfied: safetensors==0.3.1 in d:\fooocus-2\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 4)) (0.3.1)
Requirement already satisfied: accelerate==0.21.0 in d:\fooocus-2\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 5)) (0.21.0)
Requirement already satisfied: pyyaml==6.0 in d:\fooocus-2\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 6)) (6.0)
Requirement already satisfied: Pillow==9.2.0 in d:\fooocus-2\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 7)) (9.2.0)
Requirement already satisfied: scipy==1.9.3 in d:\fooocus-2\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 8)) (1.9.3)
Collecting tqdm==4.65.0 (from -r requirements_versions.txt (line 9))
  Obtaining dependency information for tqdm==4.65.0 from https://files.pythonhosted.org/packages/e6/02/a2cff6306177ae6bc73bc0665065de51dfb3b9db7373e122e2735faf0d97/tqdm-4.65.0-py3-none-any.whl.metadata
  Downloading tqdm-4.65.0-py3-none-any.whl.metadata (56 kB)
     -------------------------------------- 57.0/57.0 kB 599.2 kB/s eta 0:00:00
Requirement already satisfied: psutil==5.9.5 in d:\fooocus-2\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 10)) (5.9.5)
Requirement already satisfied: pytorch_lightning==1.9.4 in d:\fooocus-2\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 11)) (1.9.4)
Requirement already satisfied: omegaconf==2.2.3 in d:\fooocus-2\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 12)) (2.2.3)
Requirement already satisfied: gradio==3.41.2 in d:\fooocus-2\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 13)) (3.41.2)
Requirement already satisfied: pygit2==1.12.2 in d:\fooocus-2\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 14)) (1.12.2)
Requirement already satisfied: opencv-contrib-python==4.8.0.74 in d:\fooocus-2\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 15)) (4.8.0.74)
Requirement already satisfied: httpx==0.24.1 in d:\fooocus-2\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 16)) (0.24.1)
Requirement already satisfied: onnxruntime==1.16.3 in d:\fooocus-2\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 17)) (1.16.3)
Requirement already satisfied: timm==0.9.2 in d:\fooocus-2\python_embeded\lib\site-packages (from -r requirements_versions.txt (line 18)) (0.9.2)
Collecting translators==5.9.2 (from -r requirements_versions.txt (line 19))
  Obtaining dependency information for translators==5.9.2 from https://files.pythonhosted.org/packages/7b/7b/b46f279945c62421b4453e1358f3fbffa0bd3f0d693f7eaaab979deb568e/translators-5.9.2-py3-none-any.whl.metadata
  Downloading translators-5.9.2-py3-none-any.whl.metadata (68 kB)
     ---------------------------------------- 68.8/68.8 kB 1.8 MB/s eta 0:00:00
Collecting rembg==2.0.57 (from -r requirements_versions.txt (line 20))
  Obtaining dependency information for rembg==2.0.57 from https://files.pythonhosted.org/packages/0c/29/74b542899b19d33761b5fbb6a625b3dcdfd946966f7615c8d80567d29d81/rembg-2.0.57-py3-none-any.whl.metadata
  Downloading rembg-2.0.57-py3-none-any.whl.metadata (16 kB)
Collecting groundingdino-py==0.4.0 (from -r requirements_versions.txt (line 21))
  Downloading groundingdino-py-0.4.0.tar.gz (82 kB)
     -------------------------------------- 82.3/82.3 kB 924.3 kB/s eta 0:00:00
  Preparing metadata (setup.py): started
  Preparing metadata (setup.py): finished with status 'error'

stderr:   error: subprocess-exited-with-error

  python setup.py egg_info did not run successfully.
  exit code: 1

  [6 lines of output]
  Traceback (most recent call last):
    File "<string>", line 2, in <module>
    File "<pip-setuptools-caller>", line 34, in <module>
    File "C:\Users\Administrator\AppData\Local\Temp\pip-install-l99x3ycy\groundingdino-py_62cf637f2228460eabe433dd8bc89257\setup.py", line 41, in <module>
      readme = readme_file.read()
  UnicodeDecodeError: 'gbk' codec can't decode byte 0xa4 in position 2878: illegal multibyte sequence
  [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

Encountered error while generating package metadata.

See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.

[notice] A new release of pip is available: 23.2.1 -> 24.1.2
[notice] To update, run: D:\Fooocus-2\python_embeded\python.exe -m pip install --upgrade pip

CMD Failed requirements: install -r "requirements_versions.txt"
[Cleanup] Attempting to delete content of temp dir C:\Users\Administrator\AppData\Local\Temp\fooocus
[Cleanup] Cleanup successful
You do not have [realisticStockPhoto_v20.safetensors] but you have [realisticStockPhoto_v10.safetensors].
Fooocus will use [realisticStockPhoto_v10.safetensors] to avoid downloading new models, but you are not using the latest models.
Use --always-download-new-model to avoid fallback and always get new models.
Total VRAM 22528 MB, total RAM 32709 MB
Set vram state to: NORMAL_VRAM
Always offload VRAM
Device: cuda:0 NVIDIA GeForce RTX 2080 Ti : native
VAE dtype: torch.float32
Traceback (most recent call last):
  File "D:\Fooocus-2\Fooocus\entry_with_update.py", line 46, in <module>
    from launch import *
  File "D:\Fooocus-2\Fooocus\launch.py", line 144, in <module>
    from webui import *
  File "D:\Fooocus-2\Fooocus\webui.py", line 10, in <module>
    import modules.async_worker as worker
  File "D:\Fooocus-2\Fooocus\modules\async_worker.py", line 3, in <module>
    from extras.inpaint_mask import generate_mask_from_image, SAMOptions
  File "D:\Fooocus-2\Fooocus\extras\inpaint_mask.py", line 6, in <module>
    from extras.GroundingDINO.util.inference import default_groundingdino
  File "D:\Fooocus-2\Fooocus\extras\GroundingDINO\util\inference.py", line 11, in <module>
    from groundingdino.util.inference import Model
ModuleNotFoundError: No module named 'groundingdino'

thanks for your great work, i use branch"develop"

**Comment by BNP1111** [comment]
but if i run https://github.com/mashb1t/Fooocus------"develop"
it's ok

**Comment by mashb1t** [comment, marked as answer]
See https://github.com/lllyasviel/Fooocus/discussions/3282#discussioncomment-10046423
